django/contrib/gis/gdal/raster/source.py

```diff
@@ -47,23 +61,68 @@ def __init__(self, ds_input, write=False):
         self._write = 1 if write else 0
         Driver.ensure_registered()
 
+        # For JSON or HEX inputs, convert string to dict
+        if isinstance(ds_input, six.string_types) and json_regex.match(ds_input):
+            ds_input = json.loads(ds_input)
+
         # If input is a valid file path, try setting file as source.
         if isinstance(ds_input, six.string_types):
```

Comment:
I tried to use elif here, but the problem is that json inputs need some kind of pre-pocessing. These lines should check if the input string is a json. If yes, convert the input to a dict - and subsequently still use the dict option for instantiation.